src/SFA.DAS.EmployerIncentives.Events/Accounts/AccountLegalEntityRemovedHandler.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using SFA.DAS.EmployerIncentives.Abstractions.Commands;
+using SFA.DAS.EmployerIncentives.Abstractions.Events;
+using SFA.DAS.EmployerIncentives.Commands.Types.ApprenticeshipIncentive;
+using SFA.DAS.EmployerIncentives.Data.IncentiveApplication;
+using SFA.DAS.EmployerIncentives.Domain.Accounts.Events;
+
+namespace SFA.DAS.EmployerIncentives.Events.Accounts
+{
+    public class AccountLegalEntityRemovedHandler : IDomainEventHandler<AccountLegalEntityRemoved>
+    {
+        private readonly IIncentiveApplicationDataRepository _incentiveApplicationDataRepository;
+        private readonly ICommandDispatcher _commandDispatcher;
```

Comment:
Consider using the CommandPublisher.  That would put these on a queue for later processing by the Domain MessageHandlers and un-couple the two processes.